### PR TITLE
Planning scene interface: adding persistent connections for get/apply planning scene

### PR DESCRIPTION
**Pull request by CesareTonola**
The `PlanningSceneInterface` is a client for the `/get_planning_scene` and `/apply_planning_scene` services. Setting the `persistent` flag to `true` during client declaration (default is `false`) keeps the communication between the client and server open, improving performance, especially when calls to these services are repeated over time, such as in dynamic environment applications. I have introduced a new input parameter to the `PlanningSceneInterface` constructor that allows setting `persistent_connection` to `true` (default value is `false`).

### CODE REVIEW

File: moveit_ros/planning_interface/planning_scene_interface/src/planning_scene_interface.cpp [hunk 1]
```diff
@@ -80,6 +87,10 @@ class PlanningSceneInterface::PlanningSceneInterfaceImpl
     moveit_msgs::GetPlanningScene::Response response;
     std::vector<std::string> result;
     request.components.components = request.components.WORLD_OBJECT_NAMES;
+
+    if (!planning_scene_service_.isValid())
+      connectGetPlanningSceneService();
+
     if (!planning_scene_service_.call(request, response))
```
Comment (rhaschke) [inline moveit_ros/planning_interface/planning_scene_interface/src/planning_scene_interface.cpp:94]: Obviously, you need to ensure a valid connection before each call. Please encapsulate these two steps within new methods `get_planning_scene_service_call` and `apply_planning_scene_service_call`, which will clarify the interdependency.

File: moveit_ros/planning_interface/planning_scene_interface/src/planning_scene_interface.cpp [hunk 2]
```diff
@@ -312,16 +347,65 @@ class PlanningSceneInterface::PlanningSceneInterfaceImpl
...
+
+    if (wait_)
+    {
+      waitForService(planning_scene_service_);
+    }
+    else
+    {
+      if (!planning_scene_service_.exists())
+      {
+        throw std::runtime_error("ROS service not available");
```
Comment (rhaschke) [inline moveit_ros/planning_interface/planning_scene_interface/src/planning_scene_interface.cpp:368]: Throwing an exception seems not to be a good idea. Nobody catches the exception.

**Review by rhaschke — COMMENTED** [review]
This is a useful addition. Please see my comments for further improvement.

File: moveit_ros/planning_interface/planning_scene_interface/src/planning_scene_interface.cpp (hunk 1 above)
Comment (CesareTonola) [reply moveit_ros/planning_interface/planning_scene_interface/src/planning_scene_interface.cpp:94]: Ok, I would call them `getPlanningSceneServiceCall` and `applyPlanningSceneServiceCall` for consistency with the other class methods, is that ok with you?

File: moveit_ros/planning_interface/planning_scene_interface/src/planning_scene_interface.cpp (hunk 2 above)
Comment (CesareTonola) [reply moveit_ros/planning_interface/planning_scene_interface/src/planning_scene_interface.cpp:368]: I only moved what was already implemented in this [line](https://github.com/moveit/moveit/blob/9caa14eabc69c6ba2328eacfb5f6253a8acff085/moveit_ros/planning_interface/planning_scene_interface/src/planning_scene_interface.cpp#L72) in the master branch. What should I do? Should I just replace it with a return false?

File: moveit_ros/planning_interface/planning_scene_interface/src/planning_scene_interface.cpp (hunk 1 above)
Comment (rhaschke) [reply moveit_ros/planning_interface/planning_scene_interface/src/planning_scene_interface.cpp:94]: Sure.

File: moveit_ros/planning_interface/planning_scene_interface/src/planning_scene_interface.cpp (hunk 2 above)
Comment (rhaschke) [reply moveit_ros/planning_interface/planning_scene_interface/src/planning_scene_interface.cpp:368]: I see. Then keep it as is.

**Review by rhaschke — APPROVED** [review]
Thanks for iterating on this.

**Pull request merged**